doc/loganalyzer_to_pytest.md

```diff
@@ -0,0 +1,108 @@
+
+"LogProcessor" class interface:
+- Includes legacy "LogAnalyzer" class from loganalyzer.py module.
+- New:
+- extract_syslog(dst="/tmp/syslog") - Extract all syslog entries since the latest start marker and store them to "/tmp/syslog" folder on the DUT.
+
+Usage example:
+
+	def test(localhost, ansible_adhoc, testbed):
+		loganalyzer = Loganalyzer(match="{PATH}/loganalyzer_common_match.txt", expect="{PATH}/loganalyzer_common_expect.txt", ignore="{PATH}/loganalyzer_common_ignore.txt")
+		loganalyzer.init()
+		# Perform test steps
+		result = loganalyzer.analyze()
+		if result["summary"]["TOTAL MATCHES"] != 0:
+			loganalyzer.fetch_fail_artifacts(dst="test/{}".format(loganalyzer.unic_test_name))
+		pytest.fail("SOME_MESSAGE\n{}\n{}".format(result["summary"], result["match_messages"]))
+
+#### Approach 2
+#### Requirements
+Move existed loganalyzer to be supported by pytest framework.
+
+#### Overview
+At this approach all the work can be done on the Ansible host if script will download extracted syslog file to the Ansible host.
+
+Current loganalyzer implementation description:
+https://github.com/Azure/SONiC/wiki/LogAnalyzer
+
+#### Development
+Module "loganalyzer.py" with class "Loganalyzer".
+
+"Loganalyzer" class interface:
+- load_common_config() - Load regexps from common configuration files: match, expect and ignore which are located in some configuration directory or with current module.
+- get_search_regexp() - returns dictionary with defined regular expressions which are used for logs processing.
+- update_match(action="pop/add", type="match/expect/ignore", *args) - add or remove regexp for syslog processing.
+Supported parameters values:
+action: "pop", "add"
+type: "match", "expect", "ignore"
+args: *args
+- init() - Add start marker to the DUT syslog.
+- analyze() - Extract syslog and copy it to ansible host. Analyze extracted syslog files localy. Return python dictionary object.
+Return example:
+		{"summary":
+		"counters": {"match": 1, "expected_match": 0, "expected_missing_match": 0},
+		"match_files": {"/tmp/syslog": {"match": 0, "expected_match": 32},
+						"/tmp/syslog1": {"match": 0, "expected_match": 15}},
+						"match_messages": {"/tmp/syslog1": ["Message 1", "Message 2", "Message n"],
+						 "/tmp/syslog2": ["Message 1", "Message 2", "Message n"]}
+	}
+- run_cmd(callable) - call function and analyze DUT logs, return the same result as "analyze" function
+
+Usage example:
+
+	from loganalyzer import Loganalyzer
+	def test(localhost, ansible_adhoc, testbed):
+		loganalyzer = Loganalyzer()
+
+		# If it is a need to load common search regular expressions. It will load common: match, expect, ignore regular expressions.
+		loganalyzer.load_common_config()
+
+		# Add start marker to DUT syslog
+		loganalyzer.init()
+
+		# Add search marker specific for None MAC test
+		loganalyzer.update_match(action="add", type="match", "Runtime error: can't parse mac address 'None'")

```

Comment:
I don't think action='pop' is needed. I would better recreate the loganalyzer object rather than keeping in mind which matches I removed and which not

Reply:
Disagree. 1 - Probably this class will be wrapped in pytets fixture, so object recreation does not fit.
2 - We need interface to add/remove specific regexp entries to tune search, so such interface will be ok in such case.
3 - If there is a need we can add "extend" flag to "load_common_config()" which can clear config:
- if "extend is True", load common configuration and add to the existed.
- if "extend is False", clear previous configuration and load common.